qiskit/assembler/assemble_schedules.py

```diff
@@ -109,18 +107,34 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
                         channel=instruction.channels[0])
                 # add samples to pulse library
                 user_pulselib[name] = instruction.command
-            elif isinstance(instruction, AcquireInstruction):
+
+            if isinstance(instruction, AcquireInstruction):
                 max_memory_slot = max(max_memory_slot,
                                       *[slot.index for slot in instruction.mem_slots])
+                acquire_instruction_map[(shift, instruction.command)].append(instruction)
```

Comment:
I think a comment here about why this is needed would probably be pretty helpful in the future.